packages/frontend/src/features/post/components/Post/PostReportDialog.tsx

```diff
@@ -0,0 +1,150 @@
+import { Dialog } from '@/features/shared'
+import { ReportCategory } from '@/types/Report'
+import { FieldValues, UseFormHandleSubmit } from 'react-hook-form'
+import { useReportForm } from '../../hooks/useReportForm'
+import { useReportPost } from '../../hooks/useReportPost/useReportPost'
+import {
+    REGISTER_ID_DESC,
+    REGISTER_ID_REASON,
+    ReportFormCtn,
+    ReportFormDesc,
+    ReportFormIntro,
+    ReportFormReasons,
+    ReportFormStepGroup,
+    ReportFormStepLabel,
+    ReportFormSubmitBtn,
+    ReportFormSubmitFailure,
+    ReportFormSubmitSuccess,
+    ReportFormSubmitting,
+} from '../ReportForm'
+
+interface PostReportDialogProps {
+    postId: string
+    isOpen: boolean
+    onClose: () => void
+}
+
+const defaultValues = {
+    [`${REGISTER_ID_REASON}`]: -1,
+    [`${REGISTER_ID_DESC}`]: '',
+}
+
+export function PostReportDialog({
+    postId,
+    isOpen,
+    onClose,
+}: PostReportDialogProps) {
+    const {
+        handleSubmit,
+        register,
+        errors,
+        setValue,
+        getValues,
+        trigger,
+        isSubmitFailure,
+        isSubmitting,
+        isSubmitSuccess,
+        isNotSubmitted,
+        resetToCurrentState,
+        resetAll,
+        updateStateToFailure,
+        updateStateToSuccess,
+        updateStateToSubmitting,
+    } = useReportForm(defaultValues)
+
+    const onSubmit = useSubmitReportFlow({
+        postId,
+        handleSubmit,
+        updateStateToSubmitting,
+        updateStateToFailure,
+        updateStateToSuccess,
+    })
```

Comment:
Last time I get confused by similar naming between this file and react-hook-form about `isSubmitting` and so on. However, I found the purpose is different. The purpose of  react-hook-form is to make form validation easy instead of to track states of async function. So I suggest that you can use react-query to track these states and simplify the flow.

For example:
```
const { isPending, isSuccess, isError } = useMutation({...})
```
You can refer to this file. https://github.com/social-tw/social-tw-website/blob/dc2e8d9a45a89d24323a435662b7c9830d7b8111/packages/frontend/src/features/reporting/components/Adjudicate/Adjudicate.tsx#L19-L40